### PR TITLE
fix: invalidate children key on reorder

### LINKED ISSUE #358: Reorder items does not update

### PR DESCRIPTION
**Pull request by pyphilia**
close #358

### CODE REVIEW

File: src/mutations/item.test.ts [hunk 1]
```diff
@@ -257,6 +257,47 @@ describe('Items Mutations', () => {
       expect(queryClient.getQueryState(parentKey)?.isInvalidated).toBeTruthy();
     });
 
+    it.only('Edit item extra children order invalidate children key', async () => {
```
Comment (spaenleh) [inline src/mutations/item.test.ts:260]: Remove `.only`

File: src/mutations/item.ts [hunk 2]
```diff
@@ -240,6 +242,12 @@ export default (queryConfig: QueryClientConfig) => {
             queryClient.invalidateQueries(parentKey);
           }
 
+          // reorder affect children to change
+          if ((extra as FolderItemExtra)?.[ItemType.FOLDER]?.childrenOrder) {
+            console.log('woirfjkm');
```
Comment (spaenleh) [inline src/mutations/item.ts:247]: Remove ?

File: src/mutations/item.ts [hunk 3]
```diff
@@ -240,6 +242,12 @@ export default (queryConfig: QueryClientConfig) => {
             queryClient.invalidateQueries(parentKey);
           }
 
+          // reorder affect children to change
+          if ((extra as FolderItemExtra)?.[ItemType.FOLDER]?.childrenOrder) {
```
Comment (spaenleh) [inline src/mutations/item.ts:246]: Would checking the type of the item help here ?

**Review by spaenleh — CHANGES_REQUESTED** [review]

File: src/mutations/item.ts (hunk 3 above)
Comment (pyphilia) [reply src/mutations/item.ts:246]: You cannot because you don't pass the `type` when you edit an item.

**Review by spaenleh — APPROVED** [review]

**Pull request merged**